### ISSUE TITLE
Servlet 4.0 and HTTP/2 support

### ISSUE THREAD

**Issue by takezoe** [scalatra/scalatra#757]

**Comment by magnolia-k** [comment]
tomcat 9 already supports servlet 4.0, but jetty supports version 4.0 for servlet 4.0.
If jetty 10 is not released, servlet 4.0 correspondence will be difficult.

**Comment by takezoe** [comment]
@magnolia-k Thanks for advice! It should be postponed. I remove this issue from 2.7.0 milestone.

**Comment by magnolia-k** [comment]
https://github.com/eclipse/jetty.project/releases/tag/jetty-10.0.0

Jetty 10.0 has finally been released!

However, Jetty 10 supports Java 11 and above, so if you want to use Jetty 10 based code, you will need Java 11 or above for Scalatra itself.

**Comment by takezoe** [comment]
Hmm... I see. Scalatra 2.8 may be a chance to get rid of Java 8 support.

**Comment by magnolia-k** [comment]
I agree with dropping Java8 support from the new version of Scalatra.

**Comment by magnolia-k** [comment]
@takezoe 

In Jetty 10.0, the WebSocket related code has been refactored and the configuration has changed significantly.
Atmosphere does not yet support Jetty 10, so simply upgrading Jetty will cause the Atmosphere module tests to fail.
(Jetty no longer uses the org.eclipse.jetty.websocket namespace).

This means that in order to upgrade Scalatra's Jetty to 10.0, we will have to wait for Atmosphere to support Jetty 10, but at the current development speed of Atmosphere, it does not seem very likely that this will happen.

This means that we should drop support for Atmosphere in order to make Scalatra Jetty 10.0 compatible.

Also, since Jetty 11.0 changes the javax namespace to jakarta, the basic servlet api will need to be changed, and the metrics module will need to be reviewed.

So, for the time being, upgrading to Jetty 10 seems to be the way to go. s

Translated with www.DeepL.com/Translator (free version)

**Comment by takezoe** [comment]
I agree with dropping Atmosphere support. So, our next actions will be:

- Deprecate atmosphere in Scalatra 2.8
- Drop Atmosphere support and upgrade to Jetty 10 in Scalatra 2.9

It seems to be a very long way.

**Comment by magnolia-k** [comment]
Yes!
This is going to take a long time.

Now that we've created a branch for Scalatra 2.8, we'll deprecate it against Atmosphere.
We'll also add an annotation on the Scalatra-Website saying that it will be deprecated in a future version.

**Comment by joakime** [comment]
Announced: Community End of Life for Jetty 9.4.x

* https://github.com/eclipse/jetty.project/issues/7958

This is the first step, a "heads up", the next step will be a formal EOL (End of Life), likely in 6 months time.

**Comment by DavidPerezIngeniero** [comment]
I'm happy to check that this issue is being worked: https://github.com/scalatra/scalatra/commit/3977647601e1db6283d7e8d91ffe97a3b6b010f0
When it is due approx. a new version of Scalatra updating Jetty and Servlet support?

**Comment by takezoe** [comment]
https://github.com/scalatra/scalatra/pull/1632
